### PR TITLE
Feature#24 social networks functionality

### LINKED ISSUE #24: Social Network's Funcionality
# REQUIREMENTS
- Can add/update/delete each social network (FB, TW, Instagram, Lknd)
- Display each (just link)

# Acceptance Criteria
List all required acceptance criteria
- [ ] Foreign key to User's Table

### PR DESCRIPTION
**Pull request by IrenehMDev**
# Description
  - Funcionality to add/delete/update  user's social networks
  - Social Networks Migrations fixed
  - Modified Styles for Navbar

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -23,8 +23,8 @@ html {
   box-sizing: inherit;
 }
 
-.contenedor {
+.con {
   margin: 0 auto;
-  max-width: 120rem;
+  max-width: 140rem;
```
Comment (JuanequeX) [inline app/assets/stylesheets/application.scss:28]: this kind of changes are not belong to the functionality that you are solve it. please, do not mix changes that do not make sense with what you are described into the PR. 

please remove this changes and push only things relathion with your task.

**Review by LuisGomez89 — APPROVED** [review]

File: app/assets/stylesheets/components/_navbar.scss [hunk 2]
```diff
@@ -1,37 +1,46 @@
...
 
+    .flex{
+      display: flex;
+      gap: 3rem;
+    }
+    .div{
+      display: grid;
+      grid-template-columns: 4fr 2fr 1fr;
+      gap: 0.1rem;
+    }
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_navbar.scss:32]: Same here, this change are not belong to solve at social networks functionality.

this change are stylesheet, push it in a specific PR  with this kind of changes.

File: app/assets/stylesheets/components/_navbar.scss [hunk 3]
```diff
@@ -40,8 +49,9 @@
       background-color: $background_color;
       background-position: center center;
       background-repeat: no-repeat;
-      background-size: 3.5rem;
-      width: 7rem;
+      background-size: 3rem;
+      border-style: none;
+      border: 1px solid $header_color;
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_navbar.scss:54]: same here

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: app/controllers/social_networks_controller.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+class SocialNetworksController < ApplicationController
+  def new
+    @social_network = SocialNetwork.new
+  end
+
+  def create
+    hash = params.require(:social_network).permit(:url, :network)
```
Comment (YilKanda) [inline app/controllers/social_networks_controller.rb]: Try to call the params in a private method

File: app/views/shared/_navbar.html.erb [hunk 5]
```diff
@@ -1,12 +1,16 @@
 <nav class="header-nav">
-  <div class="container">
+  <div class="con">
```
Comment (YilKanda) [inline app/views/shared/_navbar.html.erb]: I recommend leaving that class as a container, remember to give css classes names more in line with their function. That will help you in the future to identify which classes you use and why.

File: db/migrate/20220211010810_create_social_networks.rb [hunk 6]
```diff
@@ -1,10 +1,8 @@
 class CreateSocialNetworks < ActiveRecord::Migration[6.1]
   def change
     create_table :social_networks do |t|
-      t.string :twitter
-      t.string :Facebook
-      t.string :LinkendIn
-      t.string :Instagram
+      t.string :url
+      t.string :network
```
Comment (YilKanda) [inline db/migrate/20220211010810_create_social_networks.rb:5]: Good change of table values, some time ago I had made this observation.

File: test/controllers/social_networks_controller_test.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class SocialNetworksControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (YilKanda) [inline test/controllers/social_networks_controller_test.rb]: Delete the code that is in comments.

File: app/views/profiles/_socialn.html.erb [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+  <div>
+    <%= link_to network.network , network.url, target: "_blank" %>
+    <% if current_user.id == @user.id %>
+      <%= link_to 'X', social_network_path(network), method: :delete ,data: {confirm: "Are you sure ?"}%>
+    <% end %>
+  </div>
+<% end %>
+<% if current_user.id == @user.id %>
+  <%= link_to 'add', new_social_network_path(current_user.id)%>
+<% end %>
```
Comment (JuanequeX) [inline app/views/profiles/_socialn.html.erb:12]: Add a blan line here

File: app/controllers/social_networks_controller.rb (hunk 4 above)
Comment (AngelBFdev) [reply app/controllers/social_networks_controller.rb]: Done

File: db/migrate/20220211010810_create_social_networks.rb (hunk 6 above)
Comment (AngelBFdev) [reply db/migrate/20220211010810_create_social_networks.rb:5]: Done

File: test/controllers/social_networks_controller_test.rb (hunk 7 above)
Comment (AngelBFdev) [reply test/controllers/social_networks_controller_test.rb]: Done

File: app/views/shared/_navbar.html.erb (hunk 5 above)
Comment (AngelBFdev) [reply app/views/shared/_navbar.html.erb]: Done

**Review by YilKanda — APPROVED** [review]

**Review by JuanequeX — APPROVED** [review]
LGTM

**Review by Villaedi — APPROVED** [review]
LGTM

**Pull request merged**